src/coreclr/jit/importercalls.cpp

```diff
@@ -3842,6 +3845,50 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
                 break;
             }
 
+            case NI_System_Threading_Volatile_Read:
+            {
+                assert((sig->sigInst.methInstCount == 0) || (sig->sigInst.methInstCount == 1));
+                var_types retType = sig->sigInst.methInstCount == 0 ? JITtype2varType(sig->retType) : TYP_REF;
```

Comment:
Is this going to suffer from the same problem as https://github.com/dotnet/runtime/issues/85906?

Reply:
hm.. it's hard to come up with a sample that would suffer from that 🤔 perhaps something like:

```csharp
public class Class1
{
    public virtual void Test() => Console.WriteLine("Test");
}

public sealed class Class2 : Class1
{
}

static void Foo(ref Class2 addr) =>
    Volatile.Read(ref addr).Test();
```
@MichalPetryka can you check codegen for `Foo` here - is Test() inlined just fine?

Reply:
> is Test() inlined just fine?

No, both current main and this don't inline it since the JIT doesn't keep the types of refs.
But if you change it to:
```cs
public class Class1
{
    public virtual void Test() => Console.WriteLine("Test");
}

public sealed class Class2 : Class1
{
}

public static Class2 c2;

static void Foo() =>
    Volatile.Read(ref c2).Test();
```
It's then inlined fine:
```asm
G_M5489_IG01:        ; func=00, offs=000000H, size=0008H, bbWeight=1, PerfScore 1.50, gcrefRegs=0000 {}, byrefRegs=0000 {}, byref, nogc <-- Prolog IG

IN000e: 000000      stp     fp, lr, [sp, #-0x10]!
IN000f: 000004      mov     fp, sp

G_M5489_IG02:        ; offs=000008H, size=0034H, bbWeight=1, PerfScore 14.50, gcrefRegs=0000 {}, byrefRegs=0000 {}, BB01 [0000], byref

IN0001: 000008      movz    x0, #0x1CB0      // data for Test:c2
IN0002: 00000C      movk    x0, #0x5000 LSL #16
IN0003: 000010      movk    x0, #0xFFBF LSL #32
IN0004: 000014      ldapr   x0, [x0]
IN0005: 000018      ldrsb   wzr, [x0]
IN0006: 00001C      movz    x0, #0x1848
IN0007: 000020      movk    x0, #0xBABE LSL #16
IN0008: 000024      movk    x0, #0xFFBE LSL #32
IN0009: 000028      movz    x1, #0x47C8      // code for System.Console:WriteLine(System.String)
IN000a: 00002C      movk    x1, #0x65B6 LSL #16
IN000b: 000030      movk    x1, #0xFFFF LSL #32
IN000c: 000034      ldr     x1, [x1]
IN000d: 000038      blr     x1

G_M5489_IG03:        ; offs=00003CH, size=0008H, bbWeight=1, PerfScore 2.00, epilog, nogc, extend

IN0010: 00003C      ldp     fp, lr, [sp], #0x10
IN0011: 000040      ret     lr
```

Reply:
Yeah looks like your PR doesn't regresses main but we might have some missing opportunities in gtGetClassHandle (not related to your PR) so it's ok to leave as is I think